Replace purrr functions with base/BBmisc

pat-s opened the pull request:
fixes #2489

larskotthoff commented:
This seems to break a few things...

mb706 commented on R/createSpatialResamplingPlots.R:
```diff
@@ -180,11 +180,12 @@ createSpatialResamplingPlots = function(task = NULL, resample = NULL, crs = NULL
   })
 
-  plot.list = flatten(plot.list.out.all)
+  plot.list = unlist(plot.list.out.all)
```
probably `recursive = FALSE`

mb706 commented on R/getResamplingIndices.R:
```diff
@@ -45,17 +44,17 @@ getResamplingIndices = function(object, inner = FALSE) {
-          flatten(
-            map(inner_inds[[z]][[u]], ~ # map over number of inner folds
-                  list(outer_inds[["train.inds"]][[z]][.x]) # the inner test.inds are a subset of the outer train.inds! That's why "train.inds" is hardcoded here
+          unlist(
```
probably should use `recursive = FALSE`, but as I say below I dont really understand what you are trying to do here.

pat-s commented:
@mb706 	Thanks for taking the time to review this. Would have been a nice review before we merged it :)

I am sometimes unsure if all your reviews are on the positive or negative side in relation to me - they sound somewhat critical from time to time ;)

I took the time and addressed all your points and learned sth - note that I am not an IT guy by education but self-trained, i.e. such kind of things can possibly happen more often.

mb706 commented:
I agree that it would be nice to have tighter code reviews, but I'm also aware of the obvious costs of time and work, and as long as the tests pass and stuff works this will probably outweigh the limited short term benefits from consistent style and idiomatic code.

Once the linter is working again it should be helpful in this regard however.

larskotthoff commented:
Thanks, merging.
